R/plot_xctr_financial.R

```diff
@@ -0,0 +1,64 @@
+#' Plot xctr data on a company or portfolio level, with financial data
+#'
+#' Plots the financial data for either a specific company or for the portfolio as a whole.
+#'
+#' @param data A data frame like [financial]
+#' @param company_name (optional) The name of the specific company to plot the
+#' financial data for. If NULL, the function will plot the financial data for the portfolio.
+#' @param mode The mode of financial data to plot.
+#' It can be one of "equal_weight", "worst_case", "best_case", or "main_activity".
+#'
+#' @return A ggplot2 object representing the financial data plot.
+#'
+#' @export
+#'
+#' @examples
+#' # Example 1: Plot financial data for a specific company
+#' plot_xctr_financial(data = financial, company_name = "peter", mode = "equal_weight")
+#'
+#' # Example 2: Plot portfolio-level financial data
+#' plot_xctr_financial(data = financial, mode = "worst_case")
+#'
+plot_xctr_financial <- function(data, company_name = NULL, mode = c("equal_weight", "worst_case", "best_case", "main_activity")) {
+  mode <- arg_match(mode)
+
+  # TODO: do we want to drop NA's everywhere silently?
+  data <- data |>
+    drop_na(-c(.data$equal_weight_finance, .data$worst_case_finance, .data$best_case_finance, .data$main_activity))
+
+  crucial <- c(
+    "main_activity",
+    "_risk_category",
+    "equal_weight_finance",
+    "worst_case_finance",
+    "best_case_finance"
+  )
+  data |> check_crucial_names(c(names(select(data, matches(crucial)))))
+
+  risk_category_var <- names(select(data, matches("_risk_category")))
+
+  data <- data |>
+    mutate(risk_category_var = factor(
+      data[[risk_category_var]],
+      levels = c("low", "medium", "high")
+    ))
+
+  y_var <- switch_mode(mode)
+  y_label <- if (is.null(company_name)) "avg_financial_value" else y_var
+
+  data <- if (is.null(company_name)) {
+    data |>
+      group_by(.data$benchmark, risk_category_var) |>
+      summarise(avg_financial_value = mean(.data[[y_var]]))
+  } else {
+    data |>
+      filter(company_name == .env$company_name)
+  }
+
+  ggplot(data, aes(x = .data$risk_category_var, y = .data[[y_label]], fill = .data$risk_category_var)) +
+    geom_bar(stat = "identity") +
+    facet_wrap(~ .data$benchmark, scales = "fixed") +
+    fill_score_colors() +
+    theme_tiltplot() +
```

Comment:
This pattern is duplicated above. Maybe you can capture the common concept in a wrapper, e.g. `barplot_score()`?

Reply:
I'll create a #TODO for now as I am struggling to put that in a little function, since the geom_bar() or facet_wrap() need to be added to the ggplot() object and it is throwing an error :(